src/daemon/dlt_daemon_client.c

```diff
@@ -119,59 +119,81 @@ static int dlt_daemon_client_send_all_multiple(DltDaemon *daemon,
                  "%s: Invalid parameters\n",
                  __func__);
         dlt_log(LOG_ERR, local_str);
-        return 0;
+        return DLT_DAEMON_ERROR_UNKNOWN;
     }
 
     temp = daemon_local->pEvent.connections;
-    temp = dlt_connection_get_next(temp, type_mask);
 
-    /* FIXME: the lock shall include the for loop as data
-     * can be affect between each iteration, but
-     * dlt_daemon_close_socket may call us too ...
-     */
-    for (j = 0; ((j < daemon_local->client_connections) && (temp != NULL)); j++)
+    //consider if the first connection is not requested type
+    if(!((1 << temp->type) & type_mask))
```

Comment:
same condition is checked inside dlt_connection_get_next. What you are trying to achieve with this change?

Reply:
I see it now, the condition is not needed.